src/script/conversation/ConversationRepository.ts

```diff
@@ -310,13 +312,130 @@ export class ConversationRepository {
     amplify.subscribe(WebAppEvents.TEAM.MEMBER_LEAVE, this.teamMemberLeave);
     amplify.subscribe(WebAppEvents.USER.UNBLOCKED, this.onUnblockUser);
     amplify.subscribe(WebAppEvents.CONVERSATION.INJECT_LEGAL_HOLD_MESSAGE, this.injectLegalHoldMessage);
+    amplify.subscribe(WebAppEvents.FEDERATION.EVENT_FROM_BACKEND, this.onFederationEvent);
 
     this.eventService.addEventUpdatedListener(this.updateLocalMessageEntity);
     this.eventService.addEventDeletedListener(this.deleteLocalMessageEntity);
 
     window.addEventListener<any>(WebAppEvents.CONVERSATION.JOIN, this.onConversationJoin);
   }
 
+  private readonly onFederationEvent = async (event: FederationEvent) => {
+    const {type, data} = event;
+
+    if (type === FEDERATION_EVENT.FEDERATION_DELETE) {
+      const {domain: deletedDomain} = data;
+
+      await this.onFederationDelete(deletedDomain);
+    }
+
+    if (type === FEDERATION_EVENT.FEDERATION_CONNECTION_REMOVED) {
+      const {domains: deletedDomains} = data;
+      await this.onFederationConnectionRemove(deletedDomains);
+    }
```

Comment:
let's convert that into a switch in case we will need to handle more events of this type in the future. 
Also we could early return in case the type has been matched

Reply:
done in https://github.com/wireapp/wire-webapp/pull/15583/commits/76d1df00d310492f36e57e1f9338a2982bc423b7